src/js/mixins/evented.js

```diff
@@ -398,6 +418,14 @@ const EventedMixin = {
    *          Whether or not the default behavior was prevented.
    */
   trigger(event, hash) {
+    validateTarget(this.eventBusEl_, this, 'trigger');
+
+    const type = event && typeof event !== 'string' ? event.type : event;
+
+    if (!isValidEventType(type)) {
```

Comment:
We never validated trigger which could get us into all kinds of strange errors in lower helper objects. Usually in `Events` or `Dom`. Validation here, along with the new errors, will make debugging issues much easier.

Reply:
I think that chances of breaking or almost zero because an error would be thrown anyway if `isValidEventType` returned `false`. The resulting error would just be less useful.